### ISSUE TITLE
Create documentation of list of valid java properties

### ISSUE THREAD

**Issue by richardatsap** [eclipse/jetty.project#3857]
I am using jetty embedded in Jenkins and I am trying to configure client certificate authentication. To achieve this it would be helpful to know which properties I can pass to the Jetty Java process (-Dorg.eclipse.jetty...). On the documentation site I only find accidental properties but not a comprehensive list. If there exists one it would be nice if the documentation would prominently point to it.

Thanks
Richard

**Comment by sbordet** [comment]
If you are using Jetty embedded you don't need system properties, you just configure everything via API.

Client certificate authentication is configured in `SslContextFactory`.

**Comment by olamy** [comment]
@richardatsap some of the configuration will probably need sone changes in Winstone library. Do you have any issue created in Jenkins project? Maybe this one is related https://issues.jenkins-ci.org/browse/JENKINS-58094?

**Comment by richardatsap** [comment]
> 
> 
> @richardatsap some of the configuration will probably need sone changes in Winstone library. Do you have any issue created in Jenkins project? Maybe this one is related https://issues.jenkins-ci.org/browse/JENKINS-58094?

Thanks for the hint. Yes, I have trouble configuring client cert auth for Jenkins as documentation is lacking. This is the case both for Jenkins as well as for Jetty. The Jetty documentation does not mention how to configure for client certs other than requesting it in the web.xml. The only knowledge I got is though Stackoverflow (https://stackoverflow.com/questions/42024458/how-to-configure-jetty-to-request-client-certificate-for-specified-resources) which requires me to provide a sslConfig file which in turn is not documented for winstone (how to provide one). However I can inject Java properties to Jetty (I think). Therefore it would be good to have a comprehensive documentation of all environment variables jetty accepts. I think this would be valuable generally even outside the context of my individual problem which is why I opened this issue.

Maybe Jetty has a helper class (or classes) which list the properties as constants? Linking prominently to these from the documentation would be already very valuable.

Kind Regards
Richard
